test/integration/graph/test_require_same_pkg_versions.py

```diff
@@ -186,3 +187,91 @@ def build(self):
     lock = c.load("wine/conan.lock")
     # Testing it doesn't crash or anything like that
     assert "gcc/1.0#616ce3babcecef39a27806c1a5f4b4ff" in lock
+
+
+def test_require_different_versions_transitive():
```

Comment:
Can this test show that `tools.graph:skip_binaries` works ok with the new trait? I think it should not affect it at all, but better check?